shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -200,34 +241,30 @@ protected void finalize() throws Throwable {
           return;
         }
 
-        handler.post(new SurfaceTextureFinalizerRunnable(id, flutterJNI));
+        handler.post(
```

Comment:
@ColdPaleLight, @jason-simmons  I was going over the PR https://github.com/flutter/engine/commit/ce8fe8bc1ecbbc8bffc5dcb22c46e6f2bfe02968

Can you comment on why [the PR](https://github.com/flutter/engine/pull/28304)  posted a task to the main looper instead of just running directly?

Reply:
Because 'unregisterTexture' must be called on the main thread. And 'finalize' method is called on 'Finalizer' thread.

Reply:
it looks like there's a race when reading `released`, or when `unregisterTexture` is called?

Reply:
Ahh, I see, here we need to make sure that the finalizer thread can see changes of  `released` value (for example: use keyword volatile). Or we can defer `released` check to the main thread. WDYT?

Reply:
sure, deferring sounds good. Also, can we add comments about what is happening here, why we need all of this? and clarify that this is a case where someone forgot to do the right thing. https://github.com/flutter/flutter/issues/88571

Reply:
The `finalize()` execution is not guaranteed. Can we consider using `ReferenceQueue` here?